### PR TITLE
Add backend APIs for github badges feature

### PR DESCRIPTION
**Pull request by RishabhJain2018**
**Description:**
This feature will enable participants to add badges to their Github code repositories for the entry on the EvalAI leaderboard and the rank on the badge will be dynamic.
Sample badge image -
Screenshot 2020-05-25 at 3 21 01 AM

This PR contains -
- [x] Backend API for fetching data for GitHub Badges.
- [x] Refactor leaderboard API code.
- [x] Add API for fetching details of the participant team in a challenge from user auth token and challenge_pk.

### CODE REVIEW

File: apps/jobs/urls.py [hunk 1]
```diff
@@ -75,4 +75,9 @@
         views.get_bearer_token,
         name="get_bearer_token",
     ),
+    url(
+        r"^phase_split/(?P<challenge_phase_split_pk>[0-9]+)/team/(?P<participant_team_pk>[0-9]+)/github_badge/$",
```
Comment (deshraj) [inline apps/jobs/urls.py:79]: lets change the url to this 
`r"^phase_splits/(?P<challenge_phase_split_pk>[0-9]+)/teams/(?P<participant_team_pk>[0-9]+)/github_badge/$"`

Comment (deshraj) [reply apps/jobs/urls.py:79]: Notice the `s` at the end for phase split and team.

File: apps/jobs/utils.py [hunk 2]
```diff
@@ -222,3 +226,142 @@ def handle_submission_rerun(submission, updated_status):
...
+
+    # Get the leaderboard associated with the Challenge Phase Split
+    leaderboard = challenge_phase_split.leaderboard
+
+    # Get the default order by key to rank the entries on the leaderboard
+    try:
+        default_order_by = leaderboard.schema["default_order_by"]
+    except KeyError:
+        response_data = {
+            "error": "Sorry, Default filtering key not found in leaderboard schema!"
```
Comment (deshraj) [inline apps/jobs/utils.py:243]: nit: please change the message to `Sorry, `default_order_by` key missing in leaderboard schema!`

File: apps/jobs/utils.py [hunk 3]
```diff
@@ -222,3 +226,142 @@ def handle_submission_rerun(submission, updated_status):
                 "submitted_image_uri"
             ]
     return message
+
+
+def calculate_distinct_sorted_leaderboard_data(
```
Comment (deshraj) [inline apps/jobs/utils.py:231]: Can you please add numpy based docstring https://numpydoc.readthedocs.io/en/latest/format.html#id8?

File: apps/jobs/views.py [hunk 4]
```diff
@@ -491,146 +492,21 @@ def leaderboard(request, challenge_phase_split_id):
...
-        sorted_leaderboard_data = sorted(
-            leaderboard_data,
-            key=lambda k: (
-                float(k["filtering_score"]),
-                float(-k["filtering_error"]),
-            ),
-            reverse=True
-            if challenge_phase_split.is_leaderboard_order_descending
-            else False,
+    if http_status_code == status.HTTP_400_BAD_REQUEST:
```
Comment (deshraj) [inline apps/jobs/views.py:505]: I think I am missing something here. If response is 400, why we want to return the data? :/

Also, can you please add a comment above this line to say what action to take when response is 400?

File: apps/jobs/views.py [hunk 5]
```diff
@@ -655,129 +531,30 @@ def get_all_entries_on_public_leaderboard(request, challenge_phase_split_pk):
...
```
Comment (deshraj) [inline apps/jobs/views.py:553]: Same. Please add a comment before this line to explain action to be taken here.

File: apps/participants/urls.py [hunk 6]
```diff
@@ -38,4 +38,9 @@
         views.participant_team_detail,
         name="get_participant_team_details",
     ),
+    url(
+        r"^challenge/(?P<challenge_pk>[0-9]+)/team_detail/$",
```
Comment (deshraj) [inline apps/participants/urls.py:42]: Let's change the url pattern to `r"^challenges/(?P<challenge_pk>[0-9]+)/team_details/$"`

File: apps/participants/views.py [hunk 7]
```diff
@@ -386,3 +395,33 @@ def remove_self_from_participant_team(request, participant_team_pk):
         if participants.count() == 0:
             participant_team.delete()
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+@api_view(["GET"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def participant_team_detail_in_challenge(request, challenge_pk):
```
Comment (deshraj) [inline apps/participants/views.py:404]: Please update the view function name accordingly to `get_participant_team_details_for_challenge`.

File: apps/jobs/views.py (hunk 4 above)
Comment (RishabhJain2018) [reply apps/jobs/views.py:505]: We want to return the error message. The variable name was wrong. I've changed it.

File: apps/jobs/views.py (hunk 5 above)
Comment (RishabhJain2018) [reply apps/jobs/views.py:553]: The action will be returned by the function.

**Pull request merged**